Framework/DataObjects/inc/MantidDataObjects/MDGridBox.tcc

```diff
@@ -1134,13 +1134,45 @@ TMDE(void MDGridBox)::integrateSphere(
     API::CoordTransform &radiusTransform, const coord_t radiusSquared,
     signal_t &signal, signal_t &errorSquared, const coord_t innerRadiusSquared,
     const bool useOnePercentBackgroundCorrection) const {
-  // We start by looking at the vertices at every corner of every box contained,
-  // to see which boxes are partially contained/fully contained.
+
+  // --------
+  // Overview
+  //   R: peak radius
+  //   r: inner radius (useful for defining peak background)
+  //   d: distance computed with radiusTransfor
+  // all distances are compared with their squared form
+  // ---------------------------------------------------------------------------
+  // We are checking all corners of each box to see if
+  //    ! all corners are within the integration volume (sphere/ellipsoid/donut)
+  //       -> we reach the base case, add box_signal and error to the container
+  //          (signal, errorSquared)
+  //    ! at least one corner is within the integration volume
+  //       -> recursively perform the same corner check for all its child boxes
+  //          util we reach the base case
+  //    ! no corner is within the integration volume
+  //       -> compute distance (d) between peak center and
+  //          [a] box center
+  //          [b] box corners
+  //          if d_min > R: box completely outside the integration volume, skip
+  //                        it and its children to save computing time
+  //          if d_max < r: box completely falls into the donut hole, skip it
+  //                        and its children to save computing time
+  //          others: recursively search the child boxes as the current box is
+  //                  partially overlapping with the integration volume, in
+  //                  otherwords, some of its child box is within the region
+  //                  (base case), and some of its child box is outside the
+  //                  integration volume.
 
   // One entry with the # of vertices in this box contained; start at 0.
   auto verticesContained = new size_t[numBoxes];
```

Comment:
Just a quick question in terms of memory handling. Is there any reason we can't tidy up a bit here and use a vector rather than `new` arrays directly. Unless we can gurantee no exceptions are thrown until the `deletes` below then we could leak memory if something throws?

For example,
```cpp
auto verticesContained = std::vector<size_t>(numBoxes, 0); 
```

should do the same as 

```cpp
auto verticesContained = new size_t[numBoxes];
memset(verticesContained, 0, numBoxes * sizeof(size_t));
```
and be safer? And then the same for all of the other bare arrays?

Reply:
Converting `verticesContained` and `boxMightTouch` to vector has zero impact, but changing `distmaxs` and `distmins` to vector leads to error for all MDGridBox related tests (see the test log associated with 067256c below).